lib/local_authorities_url_importer.rb

```diff
@@ -0,0 +1,93 @@
+require 'csv'
+
+class LocalAuthoritiesUrlImporter
+  CONTACTS_LIST_URL = "http://local.direct.gov.uk/Data/local_authority_contact_details.csv"
+
+
+  def self.import_urls
+    fh = fetch_data
+    begin
+      new(fh).import
+    ensure
+      fh.close
+    end
+  end
+
+  def self.fetch_data
+    fetch_http_to_file(CONTACTS_LIST_URL)
+  end
+
+  def self.fetch_http_to_file(url)
+    fh = Tempfile.new(%w(local_authority_data csv))
+    fh.set_encoding('ascii-8bit')
+
+    uri = URI.parse(url)
+    response = Net::HTTP.get_response(uri)
+
+    if response.code != "200"
+      raise "HTTP Fetch failed [#{response.code}]: #{url}"
+    end
+
+    fh.write response.body
+
+    fh.rewind
+    fh.set_encoding('windows-1252', 'UTF-8')
+    fh
+  end
+
+  def initialize(fh)
+    @filehandle = fh
+  end
+
+  def import
+    CSV.new(@filehandle, headers: true).each do |row|
+      begin
+        process_row(row)
+      rescue => e
+        Rails.logger.error "Error #{e.class} processing row in #{self.class}\n#{e.backtrace.join("\n")}"
+      end
+    end
+  end
+
+
+private
+
+  def process_row(row)
+    return if row['SNAC Code'].blank?
+    authority = LocalAuthority.where(snac: row['SNAC Code']).first
+    unless authority
+      Rails.logger.warn "#{self.class}: failed to find LocalAuthority with SNAC #{row['SNAC Code']}"
+      return
+    end
+    #authority.name = decode_broken_entities( row['Name'] )
```

Comment:
The Trello card considers using the names from the CSV which was why it was in there, but I'll remove it for now.  It's easy enough to add in later if necessary